### ISSUE TITLE
[BUG] Not checking for newer installers with --update

### ISSUE THREAD

**Issue by anewhouse** [grahampugh/erase-install#441]
**Describe the bug**
When -u or --update is provided and there's an existing installer, the script is not checking for a newer version

**To Reproduce**
Run `erase-install.sh --os=14 --update` to download an installer, then run the same command again.
 
**Expected behavior**
It should check to see if a newer version of the installer is available to download.

**Code/log output**
Command used: `erase-install.sh --os=14 --update`
```
2023-09-28 11:42:16 | v31.0 | [log_rotate] Finished rotating logs in /Library/Management/erase-install/log

2023-09-28 11:42:16 | v31.0 | [erase-install] v31.0 script execution started: Thu Sep 28 11:42:16 CDT 2023

2023-09-28 11:42:16 | v31.0 | [erase-install] Arguments provided: --os=14 --pkg --update
2023-09-28 11:42:16 | v31.0 | [erase-install] System version: 13.6 (Build: 22G120)
2023-09-28 11:42:16 | v31.0 | [check_for_swiftdialog_app] swiftDialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-09-28 11:42:16 | v31.0 | [check_for_swiftdialog_app] Creating dialog log (/var/tmp/dialog.2UO)...
2023-09-28 11:42:16 | v31.0 | [erase-install] Caffeinating this script (pid=79887)
2023-09-28 11:42:16 | v31.0 | [erase-install] Looking for existing installer app or pkg
2023-09-28 11:42:16 | v31.0 | [find_existing_installer] InstallAssistant package found at /Library/Management/erase-install/InstallAssistant-14.0-23A344.pkg.
2023-09-28 11:42:16 | v31.0 | [check_installer_pkg_is_valid] Checking validity of /Library/Management/erase-install/InstallAssistant-14.0-23A344.pkg.
2023-09-28 11:42:16 | v31.0 | [compare_build_versions] Comparing (1) 22G120 with (2) 23A344
2023-09-28 11:42:16 | v31.0 | [check_installer_pkg_is_valid] Installer: 23A344 >= System: 22G120 : valid build.
2023-09-28 11:42:16 | v31.0 | [erase-install] Running on architecture arm64
2023-09-28 11:42:16 | v31.0 | [erase-install] Sending quit message to dialog log (/var/tmp/dialog.2UO)
/Library/Management/erase-install/erase-install.sh:3205: no matches found: /Volumes/*macOS*
2023-09-28 11:42:16 | v31.0 | [erase-install] Cleaning working directory '/Library/Management/erase-install/content'

2023-09-28 11:42:16 | v31.0 | [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success



2023-09-28 11:42:16 | v31.0 | [finish] Script exit code: 0
```
I tested this on 31.0 and then many older versions and saw the same behavior back to 27.0. Versions 26.1 and 26.2 show the following, then it deletes the installer and tries to download another one:
```
[erase-install] Existing installer does not match requested version, so replacing..
```
Going back to 26.0 is the first time I actually saw it say "Checking for newer installer" in the log:
```
[erase-install] v26.0 script execution started: Thu Sep 28 13:01:38 CDT 2023
   [erase-install] Caffeinating this script (pid=10014)
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] Installer found at /Applications/Install macOS Sonoma.app.
   [check_installer_is_valid] Checking validity of /Applications/Install macOS Sonoma.app.
   [check_installer_is_valid] Mounting /Applications/Install macOS Sonoma.app/Contents/SharedSupport/SharedSupport.dmg
   [check_installer_is_valid] Using Build value from com_apple_MobileAsset_MacSoftwareUpdate.xml
Volume Shared Support on disk12s2 force-unmounted
   [compare_build_versions] Comparing (1) 22G120 with (2) 23A344
   [check_installer_is_valid] Installer: 23A344 >= System: 22G120 : valid build.
   [erase-install] Checking for newer installer
```

**Comment by grahampugh** [comment]
I've identified a potential reason for this, though I need to test with a valid package first so it might take a while before there's a new release. Note that this only affects the `--pkg` option, not the normal mode.

**Comment by grahampugh** [comment]
Can you try downloading the latest package at https://github.com/grahampugh/erase-install/releases/tag/v31.0? I just updated it with a potential fix.

**Comment by anewhouse** [comment]
I still see the issue happening - it occurs both when using `--pkg` and without, and when the existing installer is a .app or a .pkg. The whole script runs in less than 1 second, it doesn't even try to see if there's a new version available. I actually noticed this behavior with 27.3 also, I kept our fleet on that version while we were updating everyone to Ventura so I'm just now getting around to testing the versions that use Mist. I just assumed it was happening because I was running an older version of the script and was hoping it would be fixed when I switched the current one.
